game-core/src/main/java/games/strategy/triplea/delegate/TerritoryEffectHelper.java

```diff
@@ -81,4 +84,36 @@ private static boolean unitTypeKeepsBlitz(final UnitType type, final Territory l
     }
     return unitTypes;
   }
+
+  public static BigDecimal getMovementCost(final Territory t, final Unit unit) {
+    return getMaxMovementCost(t, Collections.singleton(unit));
+  }
+
+  /**
+   * Finds movement cost for each unit by adding 1 plus any territory effects and then returns the
+   * max movement across all units. If no territory effects then just returns the base cost of 1.
+   */
+  public static BigDecimal getMaxMovementCost(final Territory t, final Collection<Unit> units) {
+    if (getEffects(t).isEmpty()) {
+      return BigDecimal.ONE;
+    }
+    BigDecimal max = new BigDecimal(Integer.MIN_VALUE);
+    for (final Unit unit : units) {
```

Comment:
It would be interesting if over a large unit stack if there are performance savings to skip computing movement cost of the same unit type with the same movement left. I think we'd first need to know the runtime of this method with 100 or 200 units first, then it might be worth optimizing if it's a bit slower. Before any peformance optimization, or refactoring, the method should get a test so it would be safe to change.

Reply:
Yeah, I actually had thought the same thing. I decided at least for now, that it wasn't probably a major performance difference so would wait to consider it if performance seems slow.

Reply:
I agree to not do premature optimization, at the same time getting some quick perf measurements should not take too much effort. PerfTimer is pretty fast to use, have you experimented with it yet?

Reply:
So I ended up trying to use PerfTimer for some route finding tests but couldn't figure out how it works. Where does the output go? I was assuming the in game console window but didn't seem to go there. The javadocs for it seem out of date and it references a class I couldn't even find.

Reply:
It does an info logging, which is routed to console. It might be that 'debug' logging needs to be turned on in console.

Reply:
I tried switching it to debug as well.